torch/functional.py

```diff
@@ -2035,6 +2038,56 @@ def chain_matmul(*matrices, out=None):
         return _VF.chain_matmul(matrices, out=out)  # type: ignore[attr-defined]
 
 
+def cumsum(
+    self: Tensor,
+    dim: Optional[int] = None,
+    *,
+    dtype: Optional[torch.dtype] = None,
+    out: Optional[Tensor] = None,
+    axis: Optional[int] = None,
+):
+    r"""
+    cumsum(input, dim, *, dtype=None, out=None) -> Tensor
+
+    Returns the cumulative sum of elements of :attr:`input` in the dimension
+    :attr:`dim`.
+
+    For example, if :attr:`input` is a vector of size N, the result will also be
+    a vector of size N, with elements.
+
+    .. math::
+        y_i = x_1 + x_2 + x_3 + \dots + x_i
+
+    Args:
+        {input}
+        dim  (int): the dimension to do the operation over
+
+    Keyword args:
+        {dtype}
+        {out}
+
+    Example::
+
+        >>> a = torch.randint(1, 20, (10,))
+        >>> a
+        tensor([13,  7,  3, 10, 13,  3, 15, 10,  9, 10])
+        >>> torch.cumsum(a, dim=0)
+        tensor([13, 20, 23, 33, 46, 49, 64, 74, 83, 93])
+    """.format(**reduceops_common_args)
+
+    if axis is not None and dim is not None:
+        raise RuntimeError("expected either 'dim' or 'axis' to be given, not both")
+    if axis is not None:
+        dim = axis
+    if has_torch_function_unary(self):
+        return handle_torch_function(cumsum, (self,), self, dim, dtype=dtype, out=out)
+    if not torch.jit.is_scripting():
+        if torch.are_deterministic_algorithms_enabled() and self.is_cuda:
+            ref_func = importlib.import_module("torch._refs").cumsum
+            return ref_func(self, dim, dtype=dtype, out=out)
+    return _VF.cumsum(self, dim, dtype=dtype, out=out)
```

Comment:
I'm getting this mypy error:

```
Error (MYPY) [attr-defined]
  Module has no attribute "cumsum"

  >>> 2088  |    return _VF.cumsum(self, dim, dtype=dtype, out=out)
```

I'm not sure why it thinks `_VF.cumsum` doesn't exist. I guess I need to find out where type hints for `_VF` are defined and fix it

Reply:
Ah, looks like the other functions in here use `  # type: ignore[attr-defined]` to ignore this